framework/decode/file_processor.cpp

```diff
@@ -990,75 +991,30 @@ ParsedBlock BlockParser::ParseMethodCall(BlockBuffer& block_buffer)
 
-    size_t           parameter_buffer_size = static_cast<size_t>(block_header.size) - sizeof(call_id);
-    uint64_t         uncompressed_size     = 0;
-    format::HandleId object_id             = 0;
+    format::HandleId object_id = 0;
     ApiCallInfo      call_info{ GetBlockIndex() };
 
     success = success && block_buffer.Read(object_id);
     success = success && block_buffer.Read(call_info.thread_id);
 
     if (success)
     {
-        parameter_buffer_size -= (sizeof(object_id) + sizeof(call_info.thread_id));
-
-        // Optional backing store for uncompressed parameter_data, moved to ParsedBlock
-        UncompressedStore uncompressed_store;
-
-        BlockBuffer::BlockSpan parameter_data;
-        if (format::IsBlockCompressed(block_header.type))
+        const char*         label       = "method call block";
+        ParameterReadResult read_result = ReadParameterBuffer(label, block_buffer);
+        if (read_result.success)
         {
-            parameter_buffer_size -= sizeof(uncompressed_size);
-            success = block_buffer.Read(uncompressed_size);
-
-            if (success)
-            {
-                GFXRECON_CHECK_CONVERSION_DATA_LOSS(size_t, uncompressed_size);
-                const size_t data_size = static_cast<size_t>(uncompressed_size);
-                parameter_data =
-                    ReadCompressedParameterBuffer(block_buffer, parameter_buffer_size, data_size, uncompressed_store);
-                success = parameter_data.size() == data_size;
-
-                if (success)
-                {
-                    assert(parameter_data.size() == uncompressed_size);
-                    parameter_buffer_size = static_cast<size_t>(uncompressed_size);
-                }
-                else
-                {
-                    HandleBlockReadError(kErrorReadingCompressedBlockData,
-                                         "Failed to read compressed function call block data");
-                }
-            }
-            else
-            {
-                HandleBlockReadError(kErrorReadingCompressedBlockHeader,
-                                     "Failed to read compressed function call block header");
-            }
-        }
-        else
-        {
-            parameter_data = ReadParameterBuffer(block_buffer, parameter_buffer_size);
-            success        = parameter_data.size() == parameter_buffer_size;
-
-            if (!success)
-            {
-                HandleBlockReadError(kErrorReadingBlockData, "Failed to read function call block data");
-            }
-        }
-
-        if (success)
-        {
-            return ParsedBlock(
-                std::move(block_buffer.ReleaseData()),
-                MethodCallArgs{
-                    call_id, object_id, call_info, parameter_data.GetDataAs<const uint8_t>(), parameter_buffer_size },
-                std::move(uncompressed_store));
+            return ParsedBlock(std::move(block_buffer.ReleaseData()),
```

Comment:
Is `std::move` necessary here? `ReleaseData` is already returning an r-value and the same constructor is called above without `std::move`. I'm fine either way as long as we're consistent. Unless I'm missing something as to why this one is different?